zerver/lib/actions.py

```diff
@@ -2213,6 +2213,10 @@ def validate_user_access_to_subscribers_helper(user_profile: Optional[UserProfil
     if user_profile.realm_id != stream_dict["realm_id"]:
         raise ValidationError("Requesting user not in given realm")
 
+    # Guest user can access subscribed public stream's subscribers
+    if user_profile.is_guest and check_user_subscribed():
+        return
+
     if not user_profile.can_access_public_streams() and not stream_dict["invite_only"]:
```

Comment:
Do Mit zephyrs have public streams, if yes, then why don't we give the user access to subscribed public stream's subscribers here.
If this is a bug then I think this should be something like below and the above-added guest condition can be removed:
```python
 if not user_profile.can_access_public_streams() and not check_user_subscribed():
```

Reply:
Basically, the model for MIT Zephyr is that anyone can join "public" streams, but you can't see who is subscribed or browse what public streams exist.